shared/src/models/material.ts

```diff
@@ -27,6 +34,31 @@ export class Material {
     @Type(() => CanCaterFor)
     public readonly canCaterFor: CanCaterFor;
 
+    /**
+     * when the last personnel or material matching {@link maxGlobalThreshold}
```

Comment:
the last personnel or material that has a radius (either `specificThreshold` or `generalThreshold`) matching the `maxGlobalThreshold` then `maxGlobalThreshold` should be reduced to the newest element having the biggest radius.

last because there could be multiple, but it only matters if the last also gets smaller values for the two mentioned thresholds.

Could be explained better, agree